api/src/schedules/retention.ts

```diff
@@ -77,7 +79,21 @@ export async function handleRetentionJob() {
 			}
 
 			try {
-				count = await database(task.collection).where('id', 'in', subquery).delete();
+				// reset count
+				count = 0;
+
+				// subquery executed here as mysql/maria does not allow limit within an IN statement
+				const records = await subquery;
+
+				if (records.length !== 0) {
+					count = await database(task.collection)
+						.where(
+							'id',
+							'in',
+							records.map((r) => r.id),
+						)
+						.delete();
+				}
```

Comment:
Could we extract this as a generic helper (if possible) so that most vendors run the efficient single query delete?

There's also the `whereIn` function that we could use.

Reply:
> Could we extract this as a generic helper (if possible) so that most vendors run the efficient single query delete?

Updated in bdcbef4315063de36beeb6b4aec1097c0057497e
> There's also the whereIn function that we could use.

Updated in aa279d299b474b3521e4ade47c32130cc73fb52c